### DISCUSSION TITLE
Create vector fonts

### DISCUSSION THREAD

**Discussion by FabriceAuz** [micropython/micropython#9913] [Using MicroPython]
Hi
I'm using micropython on a T-Pico C3 device, equipped with a ST7789 display. I use the [st7789_mpy](https://github.com/russhughes/st7789_mpy) library, which enables to draw texts using vector fonts. Some [samples ](https://github.com/russhughes/st7789_mpy/tree/master/fonts/vector)of vector fonts are provided, but I'd like to create new ones for my application.
How can I do that? What are the available tools on Windows?
Thanks for your help.

**Comment by scruss** [comment]
The vector fonts are binary encoded renditions of the 1960s-era [Hershey](https://archive.org/details/hershey-calligraphy_for_computers) fonts. I can't immediately see how @russhughes does it.

I'm an occasional typographer. Creating a new font, even a very simple one, takes a very long time with a difficult learning curve. Is there a particular font you have in mind?

**Comment by FabriceAuz** [comment]
Thanks @scruss for your answer. Actually, no specific font in mind, I just want to use the font I have on my laptop on the ST7789 display, using the Micropython library provided by @russhughes. 

These vector fonts are very convenient as you can choose the size very easily:
`tft.draw(font, 'No Wifi', 45, 40, blue, 1)`
Here, the last argumlent is the size: if it's a little too large, I can simply change it to 
`tft.draw(font, 'No Wifi', 45, 40, blue, 0.9)`

**Reply by scruss** [reply]
Scaling is very useful. The library includes code to generate bitmap fonts from Truetype fonts, but that loses the ability to scale them
